### PR TITLE
simplify pipeline selection

### PR DESCRIPTION
**Pull request by star-e**
If `custom-pipeline` is enabled, set empty `CUSTOM_PIPELINE_NAME` to 'Builtin'.
If `custom-pipeline` is disabled, `legacy-pipeline` must be enabled.

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by yanOO1497 — APPROVED** [review]

File: cocos/game/game.ts [hunk 1]
```diff
@@ -770,20 +770,8 @@ export class Game extends EventTarget {
...
-                    if (cclegacy.rendering && !macro.CUSTOM_PIPELINE_NAME) {
-                        // set macro.CUSTOM_PIPELINE_NAME to Builtin
-                        macro.CUSTOM_PIPELINE_NAME = 'Builtin';
-                    }
-                }
-                // if custom-pipeline is not feature cropped and macro.CUSTOM_PIPELINE_NAME is not set
-                // Use legacy render pipeline
-                if (macro.CUSTOM_PIPELINE_NAME === '') {
-                    cclegacy.rendering = undefined;
+                if (cclegacy.rendering && !macro.CUSTOM_PIPELINE_NAME) {
```
Comment (minggo) [inline cocos/game/game.ts:773]: Why need to check `cclegacy.rendering`?

Comment (star-e) [reply cocos/game/game.ts:773]: `macro.CUSTOM_PIPELINE_NAME` is only used by custom pipeline.
Leaving it empty might be better when using legacy pipeline.

Comment (minggo) [reply cocos/game/game.ts:773]: Does `Builtin` mean legacy pipeline?

Comment (star-e) [reply cocos/game/game.ts:773]: `Builtin` is the official custom pipeline which is replacing the legacy forward pipeline.

Comment (star-e) [reply cocos/game/game.ts:773]: This code is saying "If custom pipeline is enabled but `CUSTOM_PIPELINE_NAME` is not set, set it to default 'Builtin'".

Comment (minggo) [reply cocos/game/game.ts:773]: Got it.

**Review by minggo — APPROVED** [review]

**Pull request merged**